P/PMIx/build_tarballs.jl

```diff
@@ -64,5 +64,10 @@ dependencies = [
     Dependency("Zlib_jll"),
 ]
 
+init_block = raw"""
+ENV["PMIX_PREFIX"] = artifact_dir
```

Comment:
What happens on platforms where we don't have artifacts?

Looks like `artifact_dir` might not be defined there?

https://github.com/JuliaParallel/PMIx.jl/runs/4357544794?check_suite_focus=true

cc: @simonbyrne

Reply:
It is included in the platform-specific code: https://github.com/JuliaBinaryWrappers/OpenMPI_jll.jl/blob/main/src/wrappers/i686-linux-gnu-libgfortran5.jl